google_fonts/lib/src/google_fonts_base.dart

```diff
@@ -125,3 +125,63 @@ String _fontStyleString(FontStyle fontstyle) {
   }
   return '';
 }
+
+class GoogleFontsFamily {
+  const GoogleFontsFamily(this.fontWeight, this.fontStyle, [this.url]);
+
+  final int fontWeight;
+  final FontStyle fontStyle;
+  final String url;
+
+  // From a string key, for example, `500italic`.
+  GoogleFontsFamily.fromString(String key, [this.url])
+      : this.fontWeight = key == _regular || key == _italic
+            ? 400
+            : int.parse(key.replaceAll(_italic, '')),
+        this.fontStyle =
+            key.contains(_italic) ? FontStyle.italic : FontStyle.normal;
+
+  GoogleFontsFamily.fromTextStyle(TextStyle style, [this.url])
+      : this.fontWeight = (style?.fontWeight?.index ?? 4) * 100,
+        this.fontStyle = style?.fontStyle ?? FontStyle.normal;
+
+  @override
+  String toString() {
+    return '$fontWeight$fontStyle';
+  }
+}
+
+// This logic is taken from the following section of the minikin library, which
+// is ultimately how flutter handles matching fonts.
+// * https://github.com/flutter/engine/blob/a5680f9388ebcef6012c446c5226a79a5ab8000c/third_party/txt/src/minikin/FontFamily.cpp#L128
+int computeMatch(GoogleFontsFamily a, GoogleFontsFamily b) {
+  if (a.fontStyle == b.fontStyle && a.fontWeight == b.fontWeight) {
```

Comment:
I think it would be better to implement equals on GoogleFontsFamily, add a TODO?

Reply:
Good call, I over-rode ==